pkg/openfeaturetest/testprovider_test.go

```diff
@@ -66,7 +64,7 @@ func TestParallelSingletonUsage(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			defer testProvider.Cleanup()
 			t.Parallel()
-			testProvider.SetProvider(t, tt.givenProvider)
+			testProvider.UsingFlags(t, tt.flags)
```

Comment:
This is the meat of the change; it's actually (very slightly) less flexible, since it means we can only use the in-memory provider, but I think it makes a lot of sense.

Instead of having users set a provider for this test scope, they set _flags_ for this test scope. IMO this is easier to reason around; people don't really care about setting a provider for testing, they care about particular flag values. Under the covers we are just using the flags to create a new memory provider and setting it to the flags in question.

Reply:
that makes sense. i like it :)